### ISSUE TITLE
False negative for ``consider-iterating-dictionary`` when using ``not`` or ``list``

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [PyCQA/pylint#5323]
### Bug description

In the following sample `a.py`, only the last if raise a `consider-iterating-dictionary`

```python
metadata = {}
if "a" not in list(metadata.keys()):
    print(1)
if "a" not in metadata.keys():
    print(1)
if "a" in list(metadata.keys()):
    print(1)
if "a" in metadata.keys():
    print(1)
```    

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:8:10: C0201: Consider iterating the dictionary directly instead of calling .keys() (consider-iterating-dictionary)

------------------------------------------------------------------
Your code has been rated at 8.89/10 (previous run: 8.89/10, +0.00)
```

### Expected behavior

All four ifs are consider-iterating-dictionary

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.5
Python 3.8.10 (default, Sep 28 2021, 16:10:42) 
[GCC 9.3.0]
```
